### DISCUSSION TITLE
Oathkeeper as load balancer in kubernetes

### DISCUSSION THREAD

**Discussion by dro-sh** [ory/oathkeeper#1015] [Q&A] [answered]
I would like use Oathkeeper as Nginx Ingress Controller. I use [helm chart](https://github.com/ory/k8s/tree/master/helm/charts/oathkeeper), set `service.proxy.type = LoadBalancer`.  
I use Docker Desktop with Kubernetes. With Nginx everything works fine: I declare Ingess manifests and can access to them. But with Oauthkeeper I only can get access on localhost. I dont understand how setup Oauthkeeper as Ingress Controller and which value use for `ingressClassName` in ingress manifests

P.S.
I have read [Othkeeper Maester](https://github.com/ory/k8s/blob/master/docs/helm/oathkeeper-maester.md) and I understand, that Maester is works like Ingress Controller. And I should use it as ingress rules. Now I dont inderstand how set `example.com` to LoadBalancer service? Also I have a physical router with real IP, then it forwards port to my PC, where I launch Docker Desktop with K8s

**Comment by Demonsthere** [comment, marked as answer]
Hello there!
Oathkeeper is a proxy, but it cannot be use as an Ingress controller, as it does not implement the kubernetes specific logic. 
What you can do is: have an existing ingress controller (like nginx or traefik), expose OAthkeeper via it, and then proxy the rest of your applications though Oathkeeper :) 

Oathkeeper-maester is a k8s controller, but it is designed to help you generate Oathkeeper rules on the fly, in a dynamic fashion,  it does not allows Oathkeeper to be a ingress controller
